float64 rad_rotate
bool clockwise